source/credentials_provider_sts_web_identity.c

```diff
@@ -0,0 +1,1120 @@
+AWS_STATIC_STRING_FROM_LITERAL(s_sts_web_identity_role_arn_config, "role_arn");
+AWS_STATIC_STRING_FROM_LITERAL(s_sts_web_identity_role_arn_env, "AWS_ROLE_ARN");
+AWS_STATIC_STRING_FROM_LITERAL(s_sts_web_identity_role_session_name_config, "role_session_name");
+AWS_STATIC_STRING_FROM_LITERAL(s_sts_web_identity_role_session_name_env, "AWS_ROLE_SESSION_NAME");
+AWS_STATIC_STRING_FROM_LITERAL(s_sts_web_identity_token_file_path_config, "web_identity_token_file");
+AWS_STATIC_STRING_FROM_LITERAL(s_sts_web_identity_token_file_path_env, "AWS_WEB_IDENTITY_TOKEN_FILE");
+
+struct aws_credentials_provider_sts_web_identity_parameters {
+    struct aws_allocator *allocator;
+    /* region is actually used to construct endpoint */
+    struct aws_byte_buf endpoint;
+    struct aws_byte_buf role_arn;
+    struct aws_byte_buf role_session_name;
+    struct aws_byte_buf token_file_path;
+};
+
+struct aws_profile_collection *s_sts_web_identity_load_profile(struct aws_allocator *allocator) {
+
+    struct aws_profile_collection *config_profiles = NULL;
+    struct aws_string *config_file_path = NULL;
+
+    config_file_path = aws_get_config_file_path(allocator, NULL);
+    if (!config_file_path) {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+            "Failed to resolve config file path during sts web identity provider initialization: %s",
+            aws_error_str(aws_last_error()));
+        goto on_error;
+    }
+
+    config_profiles = aws_profile_collection_new_from_file(allocator, config_file_path, AWS_PST_CONFIG);
+    if (config_profiles != NULL) {
+        AWS_LOGF_DEBUG(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+            "Successfully built config profile collection from file at (%s)",
+            aws_string_c_str(config_file_path));
+    } else {
+        AWS_LOGF_ERROR(
+            AWS_LS_AUTH_CREDENTIALS_PROVIDER,
+            "Failed to build config profile collection from file at (%s) : %s",
+            aws_string_c_str(config_file_path),
+            aws_error_str(aws_last_error()));
+        goto on_error;
+    }
+
+    aws_string_destroy(config_file_path);
+    return config_profiles;
+
+on_error:
+    if (config_file_path) {
+        aws_string_destroy(config_file_path);
+    }
+    if (config_profiles) {
+        aws_profile_collection_destroy(config_profiles);
+    }
+    return NULL;
+}
+
+static struct aws_byte_cursor s_default_profile_name_cursor = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("default");
+static struct aws_byte_cursor s_default_region_cursor = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("us-east-1");
+static struct aws_byte_cursor s_sts_cursor = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("sts.");
+static struct aws_byte_cursor s_amazonaws_cursor = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL(".amazonaws.com");
+static struct aws_byte_cursor s_cn_cursor = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL(".cn");
+
+struct aws_credentials_provider_sts_web_identity_parameters *s_sts_web_identity_parameters_new(
```

Comment:
Seems like, even post-refactoring as suggested below, there's still more room for breaking this function up into more manageable chunks.